pkg/networkservice/common/refresh/client_test.go

```diff
@@ -302,6 +325,82 @@ func TestRefreshClient_NoRefreshOnFailure(t *testing.T) {
 	require.Never(t, cloneClient.validator(2), testWait, testTick)
 }
 
+func TestRefreshClient_CalculatesShortestTokenTimeout(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	testData := []struct {
+		Chain                     []time.Duration
+		ExpectedRefreshTimeoutMax time.Duration
+		ExpectedRefreshTimeoutMin time.Duration
+	}{
+		{
+			Chain:                     []time.Duration{time.Hour},
+			ExpectedRefreshTimeoutMax: 20*time.Minute + time.Second,
+			ExpectedRefreshTimeoutMin: 20*time.Minute - time.Second,
+		},
+		{
+			Chain:                     []time.Duration{time.Hour, 3 * time.Minute},
+			ExpectedRefreshTimeoutMax: 54*time.Second + time.Second/2.,
+			ExpectedRefreshTimeoutMin: 54*time.Second - time.Second/2.,
+		},
+		{
+			Chain:                     []time.Duration{time.Hour, 5 * time.Second, 3 * time.Minute},
+			ExpectedRefreshTimeoutMax: 5*time.Second/3. + time.Second/3.,
+			ExpectedRefreshTimeoutMin: 5*time.Second/3. - time.Second/3.,
+		},
+		{
+			Chain:                     []time.Duration{200 * time.Millisecond, 1 * time.Minute, 100 * time.Millisecond, time.Hour},
+			ExpectedRefreshTimeoutMax: 100*time.Millisecond/3. + 30*time.Millisecond,
+			ExpectedRefreshTimeoutMin: 100*time.Millisecond/3. - 30*time.Millisecond,
+		},
+	}
+
+	timeNow, err := time.Parse("2006-01-02 15:04:05", "2009-11-10 23:00:00")
+	require.NoError(t, err)
+
```

Comment:
`time.Date()` is not returning error.

```suggestion
	timeNow := time.Date(/*todo*/)

```

Reply:
Done